R/run.R

```diff
@@ -742,16 +770,23 @@ CmdStanProcs <- R6::R6Class(
           if (self$proc_state(id) == 2.5 && grepl("Exception:", line, fixed = TRUE)) {
             self$set_proc_state(id, new_state = 3)
           }
+          if (grepl("Iteration:", line, perl = TRUE)) {
+            regex_match = regexpr('[0-9]+\\s+/',line)
+            private$iter_done_[[id]] = as.numeric(substr(line,regex_match,regex_match+attr(regex_match,'match.length')-3))
+            ignore_line = TRUE
```

Comment:
```suggestion
            regex_match <- regexpr('[0-9]+\\s+/',line)
            private$iter_done_[[id]] <- as.numeric(substr(line, regex_match, regex_match + attr(regex_match, 'match.length') - 3))
            ignore_line <- TRUE
```

Also, the line starting `as.numeric` is a bit long and hard to read. Maybe break it up into two steps, e.g.

```r
iter_done <- substr(line, regex_match, regex_match + attr(regex_match, 'match.length') - 3)
private$iter_done_[[id]] <- as.numeric(iter_done)
```

Reply:
gotcha, will do the as.numeric in a separate commit.